### PR TITLE
docs: improve api doc

### PR DESCRIPTION
**Pull request by weichweich**
## improve API documentation

fixes KILTprotocol/ticket/issues/485

## How to test:

```
yarn build:docs
```

## Checklist:

- [x] I have verified that the code works
- [x] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [x] I have [left the code in a better state](https://deviq.com/boy-scout-rule/)
- [x] I have documented the changes (where applicable)

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -24,13 +24,14 @@
...
   },
   "include": [
-    "./src/*.ts",
     "./src/types/*.ts",
     "src/wasm/*.js",
-    "src/attestation/Attester.ts",
-    "src/claim/Claimer.ts",
-    "src/verification/Verifier.ts",
-    "src/Teardown.js"
+    "src/Teardown.js",
```
Comment (wischli) [inline tsconfig.json:29]: I don't remember how this got here, but we don't have a _Teardown.js_ file 😅

File: typedoc.js [hunk 2]
```diff
@@ -1,25 +1,8 @@
...
-  exclude: ['**/*spec.ts', './src/index.ts', './src/wasm/wasm_exec.js'],
+  exclude: ['**/testSetup/*', '**/*.spec.ts', './src/wasm/wasm_exec.js'],
   listInvalidSymbolLinks: true,
-  stripInternal: true,
+  excludeExternals: true,
   excludeNotExported: true,
   excludePrivate: true,
-  hideGenerator: true,
-  mode: 'file',
-  tsconfig: 'tsconfig.json',
```
Comment (wischli) [inline typedoc.js:9]: Why remove the explicit setting of the _tsconfig.ts_?

File: typedoc.js [hunk 3]
```diff
@@ -1,25 +1,8 @@
 module.exports = {
-  exclude: ['**/*spec.ts', './src/index.ts', './src/wasm/wasm_exec.js'],
+  exclude: ['**/testSetup/*', '**/*.spec.ts', './src/wasm/wasm_exec.js'],
   listInvalidSymbolLinks: true,
-  stripInternal: true,
+  excludeExternals: true,
   excludeNotExported: true,
   excludePrivate: true,
-  hideGenerator: true,
```
Comment (wischli) [inline typedoc.js:7]: Why remove hiding the generator?

File: typedoc.js [hunk 4]
```diff
@@ -1,25 +1,8 @@
...
-  mode: 'file',
-  tsconfig: 'tsconfig.json',
-  readme: 'README.md',
```
Comment (wischli) [inline typedoc.js:10]: Why remove the Readme?

File: typedoc.js (hunk 2 above)
Comment (weichweich) [reply typedoc.js:9]: not sure? Why did we add it? It's not in the SDK. But maybe I add it again. Not sure what the best way would be... 32ea26a5317cd917e5a3dae7130f647204410f84

File: typedoc.js (hunk 3 above)
Comment (weichweich) [reply typedoc.js:7]: why would we hide it? I'm always a bit annoyed if people hide this kind of info. If I see the documentation of an open source project, I would like to know which tools they use. I don't see a point in hiding this.

Comment (wischli) [reply typedoc.js:7]: We are hiding it in the SDK but I agree with you.

File: typedoc.js (hunk 2 above)
Comment (wischli) [reply typedoc.js:9]: I guess it does not matter but it should not hurt to set it.

File: typedoc.js (hunk 4 above)
Comment (weichweich) [reply typedoc.js:10]: I thought it was redundant. The readme is already in the github project. We can add it back if you want.

**Pull request merged**